IBAnimatable/Animatable.swift

```diff
@@ -81,121 +73,45 @@ public extension Animatable where Self: UIView {
-    case .slideInLeft:
-      slideInLeft(completion)
-    case .slideInRight:
-      slideInRight(completion)
-    case .slideInDown:
-      slideInDown(completion)
-    case .slideInUp:
-      slideInUp(completion)
-    case .slideOutLeft:
-      slideOutLeft(completion)
-    case .slideOutRight:
-      slideOutRight(completion)
-    case .slideOutDown:
-      slideOutDown(completion)
-    case .slideOutUp:
-      slideOutUp(completion)
-    case .squeezeInLeft:
-      squeezeInLeft(completion)
-    case .squeezeInRight:
-      squeezeInRight(completion)
-    case .squeezeInDown:
-      squeezeInDown(completion)
-    case .squeezeInUp:
-      squeezeInUp(completion)
-    case .squeezeOutLeft:
-      squeezeOutLeft(completion)
-    case .squeezeOutRight:
-      squeezeOutRight(completion)
-    case .squeezeOutDown:
-      squeezeOutDown(completion)
-    case .squeezeOutUp:
-      squeezeOutUp(completion)
-    case .fadeIn:
-      fadeIn(completion)
-    case .fadeOut:
-      fadeOut(completion)
-    case .fadeOutIn:
-      fadeOutIn(completion)
-    case .fadeInOut:
-      fadeInOut(completion)
-    case .fadeInLeft:
-      fadeInLeft(completion)
-    case .fadeInRight:
-      fadeInRight(completion)
-    case .fadeInDown:
-      fadeInDown(completion)
-    case .fadeInUp:
-      fadeInUp(completion)
-    case .fadeOutLeft:
-      fadeOutLeft(completion)
-    case .fadeOutRight:
-      fadeOutRight(completion)
-    case .fadeOutDown:
-      fadeOutDown(completion)
-    case .fadeOutUp:
-      fadeOutUp(completion)
-    case .squeezeFadeInLeft:
-      squeezeFadeInLeft()
-    case .squeezeFadeInRight:
-      squeezeFadeInRight()
-    case .squeezeFadeInDown:
-      squeezeFadeInDown()
-    case .squeezeFadeInUp:
-      squeezeFadeInUp()
-    case .squeezeFadeOutLeft:
-      squeezeFadeOutLeft()
-    case .squeezeFadeOutRight:
-      squeezeFadeOutRight()
-    case .squeezeFadeOutDown:
-      squeezeFadeOutDown()
-    case .squeezeFadeOutUp:
-      squeezeFadeOutUp()
-    case .zoomIn:
-      zoomIn(completion)
-    case .zoomOut:
-      zoomOut(completion)
+  public func animate(_ animation: AnimationType? = nil, completion: AnimatableCompletion? = nil) {
+    switch animation ?? animationType {
+    case let .slide(way, direction):
+      slide(way: way ?? .in, direction: direction ?? .right, completion: completion)
+    case let .squeeze(way, direction):
+      squeeze(way: way ?? .in, direction: direction ?? .right, completion: completion)
+    case let .squeezeFade(way, direction):
+      squeezeFade(way: way ?? .in, direction: direction ?? .right, completion: completion)
+    case let .slideFade(way, direction):
+      slideFade(way: way  ?? .in, direction: direction ?? .right, completion: completion)
+    case let .fade(way):
+      fade(way: way ?? .in, completion: completion)
+    case let .zoom(way):
+      zoom(way: way ?? .in, completion: completion)
     case .shake:
       shake(completion)
-    case .zoomInvertIn:
```

Comment:
yes, definitely. My interrogation is about the API : 
maybe add it to Zoom : 
`zoom(.in, .shrinking)`
`zoom(.in, .magnifying)`
Or simpler : 
`zoomInvert(.in)`

What do you prefer guys ?